aea/helpers/dialogue/base.py

```diff
@@ -182,63 +238,92 @@ def outgoing_extend(self, message: "Message") -> None:
 
     def incoming_extend(self, message: "Message") -> None:
         """
-        Extend the list of messages which keeps track of incoming messages.
+        Extend the list of incoming messages with 'message'
 
-        :param messages: a message to be added
+        :param message: a message to be added
         :return: None
         """
         self._incoming_messages.extend([message])
 
+    def outgoing_safe_extend(self, message: "Message") -> bool:
+        """
+        Extend the list of outgoing messages with 'message', if 'message' is valid
+
+        :param message: a message to be added
+        :return: True if message successfully added, false otherwise
+        """
+        if self.is_valid_next_message(message):
+            self._outgoing_messages.extend([message])
+            return True
+        else:
+            return False
+
+    def incoming_safe_extend(self, message: "Message") -> bool:
+        """
+        Extend the list of incoming messages with 'message', if 'message' is valid
+
+        :param message: a message to be added
+        :return: True if message successfully added, false otherwise
+        """
+        if self.is_valid_next_message(message):
+            self._incoming_messages.extend([message])
+            return True
+        else:
+            return False
+
+    @abstractmethod
+    def is_valid_next_message(self, message: "Message") -> bool:
+        """
+        Check whether 'message' is a valid next message in this dialogue.
+
+        :param message: the message to be validated
+        :return: True if yes, False otherwise.
+        """
+
 
 class Dialogues:
-    """The dialogues class keeps track of all dialogues."""
+    """The dialogues class keeps track of all dialogues for an agent."""
 
-    def __init__(self) -> None:
+    def __init__(self, agent_address: Address) -> None:
         """
         Initialize dialogues.
 
+        :param agent_address: the address of the agent for whom dialogues are maintained
         :return: None
         """
         self._dialogues = {}  # type: Dict[DialogueLabel, Dialogue]
+        self._agent_address = agent_address
         self._dialogue_nonce = 0
 
     @property
     def dialogues(self) -> Dict[DialogueLabel, Dialogue]:
-        """Get dictionary of dialogues in which the agent is engaged in."""
+        """Get dictionary of dialogues in which the agent engages."""
         return self._dialogues
 
-    @abstractmethod
-    def is_permitted_for_new_dialogue(self, msg: Message) -> bool:
-        """
-        Check whether an agent message is permitted for a new dialogue.
-
-        :param msg: the agent message
-
-        :return: a boolean indicating whether the message is permitted for a new dialogue
-        """
+    @property
+    def agent_address(self) -> Address:
+        """Get the address of the agent for whom dialogues are maintained."""
+        return self._agent_address
```

Comment:
Add setter and getter for counterparty here as well.

Reply:
On second thought, counterparts here does not work, as different dialogues may be with different counterparts. 

We can add agent_address and counterparts properties for dialogue, but I suggest we hold that off until these field are needed.

Reply:
I think my comment was referring to dialogue not dialogues...